File: Go/1456. Maximum Number of Vowels in a Substring of Given Length.go
package main

//goland:noinspection GoUnusedFunction
func maxVowels(s string, k int) int {
	maxVowel := 0
	vowels := map[byte]bool{
		'a': true, 'e': true, 'i': true, 'o': true, 'u': true,
	}

	for i := 0; i < k; i++ {
		if vowels[s[i]] {
			maxVowel++
		}
	}

	current := maxVowel
	for i := k; i < len(s); i++ {

		if vowels[s[i-k]] {
			current--
		}

		if vowels[s[i]] {
			current++
		}

		if current > maxVowel {
			maxVowel = current
		}
	}

	return maxVowel
}

File: PHP/1456. Maximum Number of Vowels in a Substring of Given Length.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    public function maxVowels(string $s, int $k): int
    {
        $max = 0;
        $vowels = [
            'a' => true, 'e' => true, 'i' => true, 'o' => true, 'u' => true,
        ];

        for ($i = 0; $i < $k; $i++) {
            if (array_key_exists($s[$i], $vowels)) {
                $max++;
            }
        }

        $current = $max;
        for ($i = $k; $i < strlen($s); $i++) {
            if (array_key_exists($s[$i - $k], $vowels)) {
                $current--;
            }

            if (array_key_exists($s[$i], $vowels)) {
                $current++;
            }

            if ($current > $max) {
                $max = $current;
            }
        }

        return $max;
    }
}
